From 8a1339e38380c2cdb5b848e2746422064b0e55bf Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Wed, 11 Sep 2019 22:58:18 +0100
Subject: [PATCH] Ticket #3629: configure.ac: drop bundled gettext

Bundled libintl did not support linking to internal static
libraries (libmc in our case): directly specified static
libraries are not pulled by libtool and are not usable for
dynamic libraries as PIC-related flags are not passed for
compilation.

This renders bundled libintl library unusable.

The change drops libintl bundling support and always relies
on external libintl (or falls back to disabled NLS).

On a related note gettext-0.20 drops support for bundling
or libintl and this change will ease migration to newer version.

The change is tested on x86_64-gentoo-linux-musl: mc builds
and links all tests successfully. A few tests fail for lack
of NLS support.

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 .gitignore            | 1 -
 Makefile.am           | 2 +-
 configure.ac          | 3 +--
 doc/doxygen.cfg       | 2 +-
 lib/Makefile.am       | 2 +-
 m4.include/mc-i18n.m4 | 5 -----
 6 files changed, 4 insertions(+), 11 deletions(-)

diff --git a/.gitignore b/.gitignore
index 488ed421f..67f81ead2 100644
--- a/.gitignore
+++ b/.gitignore
@@ -20,7 +20,6 @@ config.sub
 configure
 configure.mc
 gettext.m4
-intl
 install-sh
 m4
 mc.qpg
diff --git a/Makefile.am b/Makefile.am
index ac05a83ea..f86f6ed38 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,7 @@
 ## Process this file with automake to create Makefile.in.
 AUTOMAKE_OPTIONS = 1.5
 
-SUBDIRS = intl po lib src doc contrib misc
+SUBDIRS = po lib src doc contrib misc
 
 if HAVE_TESTS
     SUBDIRS += tests
diff --git a/configure.ac b/configure.ac
index f516048a6..79b63b39f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -272,7 +272,7 @@ dnl ############################################################################
 dnl Internationalization
 dnl ############################################################################
 
-AM_GNU_GETTEXT([no-libtool], [need-ngettext])
+AM_GNU_GETTEXT([external], [need-ngettext])
 AM_GNU_GETTEXT_VERSION([0.18.1])
 
 mc_I18N
@@ -675,7 +675,6 @@ doc/hlp/pl/Makefile
 doc/hlp/ru/Makefile
 doc/hlp/sr/Makefile
 
-intl/Makefile
 po/Makefile.in
 ])
 
diff --git a/doc/doxygen.cfg b/doc/doxygen.cfg
index 07bc973ae..1118062f6 100644
--- a/doc/doxygen.cfg
+++ b/doc/doxygen.cfg
@@ -91,7 +91,7 @@ FILE_PATTERNS          = *.c \
 RECURSIVE              = YES
 EXCLUDE                =
 EXCLUDE_SYMLINKS       = NO
-EXCLUDE_PATTERNS       = */intl/* */tests/* */.git/*
+EXCLUDE_PATTERNS       = */tests/* */.git/*
 EXCLUDE_SYMBOLS        =
 EXAMPLE_PATH           = $(SRCDIR)
 EXAMPLE_PATTERNS       =
diff --git a/lib/Makefile.am b/lib/Makefile.am
index c448e2ddb..69c47601f 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -74,4 +74,4 @@ else
     libmc_la_LIBADD += $(GLIB_LIBS)
 endif
 
-libmc_la_LIBADD += $(PCRE_LIBS) $(LIBICONV) $(LIBINTL)
+libmc_la_LIBADD += $(PCRE_LIBS) $(LIBICONV)
diff --git a/m4.include/mc-i18n.m4 b/m4.include/mc-i18n.m4
index 724ca5d26..8f1474139 100644
--- a/m4.include/mc-i18n.m4
+++ b/m4.include/mc-i18n.m4
@@ -8,11 +8,6 @@ dnl @license GPL
 dnl @copyright Free Software Foundation, Inc.
 
 AC_DEFUN([mc_I18N],[
-
-    if test "x$USE_INCLUDED_LIBINTL" = xyes; then
-        CPPFLAGS="$CPPFLAGS -I\$(top_builddir)/intl -I\$(top_srcdir)/intl"
-    fi
-
     dnl User visible support for charset conversion.
     AC_ARG_ENABLE([charset],
         AS_HELP_STRING([--enable-charset], [Support for charset selection and conversion @<:@yes@:>@]))
-- 
2.23.0

